### PR TITLE
Improve the configuration time if the build

### PR DESCRIPTION
**Pull request by alpar-t**
This will help with reproduction lines and running tests form IDEs and
other operations that are quick and executed often enough for the
configuration time to matter.

Running Gradle with a FIPS JVM is not supproted, so if the runtime JVM
is the same one, no need to spend time checking for fips support.

Verification of the JAVA<version>_HOME env vars is now async and
parallel so it doesn't block configuration.

### CODE REVIEW

**Review by mark-vieira — APPROVED** [review]
Funny that I also looked at improving this recently. My approach is a bit more involved, as I've actually completely moved all this work to a task at execution time that has the benefit that we can then cache the results with the build cache. In any case, there a bit more work required there so I think this is a very good immediate improvement to not block during configuration time on this stuff. 👍

File: buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy [hunk 1]
```diff
@@ -190,24 +197,38 @@ class BuildPlugin implements Plugin<Project> {
...
-                    final String message =
-                            "the environment variable JAVA" + version + "_HOME must be set to a JDK installation directory for Java" +
-                                    " ${expectedJavaVersionEnum} but is [${javaHome}] corresponding to [${javaVersionEnum}]"
-                    throw new GradleException(message)
+            ExecutorService exec = Executors.newFixedThreadPool(javaVersions.size())
+            Set<Future<Void>> results = new HashSet<>()
+
+            javaVersions.entrySet().stream()
+                    .filter { it.getValue() != null }
+                    .forEach { javaVersionEntry ->
```
Comment (rjernst) [inline buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy]: pet peeve: I find this use of forEach to be counter to the intention of forEach being added to java. I think this should be a loop, as it was before. Streaming with forEach is for doing simple operations on a set of data, while this has an entire loop body embedded in a lambda, making the indentation much deeper than it was originally, and thus not as easily inspectable in an editor.

**Review by rjernst — COMMENTED** [review]
This look ok, but I have one issue with this that I want to make a point of now, as I have seen a lot of code converted to "streaming loops" recently.

File: buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy (hunk 1 above)
Comment (atorok) [reply buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy]: @rjernst dully noted. I personally value having the filters before and outside the body of the loop, but don't have a strong preference. The main reason for converting this was that it had a `.parallel()` at some point that I replaced with the executor eventually.

**Comment by atorok** [comment]
@mark-vieira 
Having these as a task would be better indeed,  that was my first taught as well and went with this because we can get some immediate benefits.  I wasn't sure how best to add the task to the execution graph. Do we add a dependsOn for each task that uses it ? Some of these setting are tracking the tasks that use them to say that these tasks require that env var. We are not very consistent about it,  but I think it's something that would be nice to keep. 
Maybe have some DSL to offer a nice way to say that some task depends on "something"  - that could be some information, like a runtime java path that may need some verification, or some setup, potentially a java version that we download and set up as part of the build. 
Then we wouldn't have to pass this information around in ext properties either.

**Comment by mark-vieira** [comment]
> Do we add a dependsOn for each task that uses it ?

The plugin simply adds a dependency to the task to every other task in the project. So no change in behavior really. Now this _always_ runs because it runs at configuration time. In my refactor it also always runs for any build execution because every task depends on the task that does so.

**Pull request merged**